rust/agents/relayer/src/msg/gas_payment/mod.rs

```diff
@@ -44,10 +44,13 @@ pub struct GasPaymentEnforcer {
 }
 
 impl GasPaymentEnforcer {
-    pub fn new(
-        policy_configs: impl IntoIterator<Item = GasPaymentEnforcementConf>,
-        db: HyperlaneRocksDB,
-    ) -> Self {
+    pub fn new(policy_configs: Vec<GasPaymentEnforcementConf>, db: HyperlaneRocksDB) -> Self {
```

Comment:
If I read https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/af5bd88aa86bb1983a6f169c82755db854e8a1a6/rust/agents/relayer/src/settings/mod.rs#L190 correctly, it actually already defaults to None?

Reply:
In which case this PR is superflous?

Reply:
that defaults to an empty vec rather than `vec![(Box::new(GasPaymentPolicyNone), MatchingList::default())]`

Because of that, `policies` [here](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/302c8e471a21359fec9300e6d5b11ae31526dd89/rust/agents/relayer/src/msg/gas_payment/mod.rs#L90-L91) would be empty and you would end up seeing this error: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/302c8e471a21359fec9300e6d5b11ae31526dd89/rust/agents/relayer/src/msg/gas_payment/mod.rs#L124-L125